.github/workflows/build_and_deploy.yml

```diff
@@ -12,12 +12,35 @@ on:
 
 jobs:
   call_build:
-    uses: yext/answers-search-ui/.github/workflows/build.yml@dev/build-workflow
+    uses: yext/answers-search-ui/.github/workflows/build.yml@feature/gh-action-migration
```

Comment:
I only thought of this now, but what are the benefits of downloading the build assets instead of having each workflow individually build them? It definitely feels smarter, and it probably uses overall less computation, but since github actions are completely free the only benefit I could see is the added flexibility if we wanted to make the SDK private? Does having a separate build workflow let us just have one file to put optimizations and what not in?

Reply:
Ya, I think you listed most of the important reasons. It's DRY, and offers strong separation of concerns. Changes to the build happen in one spot. With just this one workflow (build_and_deploy), it's not necessary. However separating these out will make it super easy to create the i18n, search bar, and the versioned release workflows.

Reply:
While it is free, the max number of concurrent jobs for enterprise is 180, so minimizing computation done helps keep us under that. Extra jobs get put in queue. [Source](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration)